.github/rawWorkflows/gh-ci-parameterized-flow.txt

```diff
@@ -28,3 +28,16 @@
         uses: gradle/gradle-build-action@v2
         with:
           arguments: "$GradleArguments"
+      - name: Package Build Artifacts
+        shell: bash
+        run: |
+          mkdir ${{ github.job }}-artifacts
+          find . -path "**/build/reports/*" -or -path "**/build/test-results/*" > artifacts.list
+          rsync -R --files-from=artifacts.list . ${{ github.job }}-artifacts
+          tar -zcvf ${{ github.job }}-artifacts.tar.gz ${{ github.job }}-artifacts
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.job }}
+          path: ${{ github.job }}-artifacts.tar.gz
+          retention-days: 30
```

Comment:
When the job fails, the artifacts won't get uploaded. Should we split `Package Build Artifacts` into two separate jobs?

One that adds the coverage files that always get uploaded and one for the TestNG results if the preceding job fails.

Reply:
> When the job fails, the artifacts won't get uploaded.

I think you're right. 

> One that adds the coverage files that always get uploaded and one for the TestNG results if the preceding job fails.

I think adding `if: success() || failure()` in `Package Build Artifacts` and `if: success()` in `Upload Build Artifacts` should be enough as we've decided to upload logs always.  Besides IMO both coverage and TestNG results are sort of dependent so not sure if that will help.

Reply:
I thought we want to upload test logs only if the tests fail. We want the coverage for every execution

Reply:
As discussed offline we'll upload artifacts always. If we hit any issues we'll fix them forward. Thanks!